### ISSUE TITLE
Make sure <style> is present when editor is created

### ISSUE THREAD

**Issue by ostrolucky** [ckeditor/ckeditor5#960]
## Is this a bug report or feature request? (choose one)

🐞 Bug report  |  🆕 Feature request  |  Other

## 💻 Version of CKEditor

1.0.0-beta.3

## 📋 Steps to reproduce

```html
<html>
	<head>
		<meta charset="utf-8">
		<title>CKEditor</title>
		<script src="https://cdn.ckeditor.com/ckeditor5/1.0.0-beta.3/classic/ckeditor.js"></script>
	</head>
	<body>
		<textarea name="content" id="editor">This is some sample content.</textarea>
		<script>
			ClassicEditor.create( document.querySelector( '#editor' ) )
       setTimeout(() => {
         document.querySelectorAll('style').forEach(elem => elem.remove())
         ClassicEditor.create(document.querySelector('#editor'))
       }, 1000);
		</script>
	</body>
</html>
```
or https://codepen.io/anon/pen/EEzKqy
## ✅ Expected result
Editor should be re-rendered correctly when `.create` is called

## ❎ Actual result
Stylesheet is broken, huge fonts are present after removing <style>

## 📃 Other details that might be useful
<style> is being removed and then replaced in actual use case by 3rd party javascript library. Would be nice if this was fixed, or if there was an option to handle this on our side. We can fire some callback before removal of these styles triggered

**Comment by jodator** [comment]
> <style> is being removed and then replaced in actual use case by 3rd party javascript library

Just out of curiosity what's that 3rd party JS library? Removing `<style>` tag looks like a very aggressive move and frankly I'm not sure if we can do much about it. A similar case would be with some 3rd party JS library that removes every `<button>` tag from a webpage and thus removes all buttons from CKEditor's toolbar. Hence it would be better to know something more about that actual use case - why that lib is removing all the `<style>` tags?

**Comment by ostrolucky** [comment]
It's SPA-like library [ajaxcom-js](https://github.com/everlutionsk/ajaxcom-js). Similar to turbolinks and Pjax.

It binds itself to all links and retrieves container content via http fetch. It then replaces container content, script and styles with the ones retrieved. This allows browser to skip re-rendering unnecessary parts. This works for CKeditor too, but only for first time. After that ckeditor always wrongly assumes it's <style> tag is already there.

> A similar case would be with some 3rd party JS library that removes every <button> tag from a webpage and thus removes all buttons from CKEditor's toolbar.

Yes, that's desired thing though. Point IS to remove it and I don't expect CKEditor to work. What I expect to work is when elements are re-added and ckeditor is created again. So in given example button is removed, but then readded back and events re-binded. It should work. I haven't yet encountered library with such a strong global state/singleton like CKEditor 5 have with it's <style> handling.

**Comment by jodator** [comment]
OK now I understand this a bit more. We're using [Webpack plugin](https://github.com/webpack-contrib/style-loader) to build and bundle CSS into `ckeditor.js`.
What you need is custom [build that extracts CSS](https://docs.ckeditor.com/ckeditor5/latest/builds/guides/integration/advanced-setup.html#option-extracting-css) so it can be re-added later. 

It will create a `styles.css` file alongside `ckeditor.js` which you should add this on every `ClassicEditor.create()` call.

Unfortunatelly the [CDN builds](https://cdn.ckeditor.com/#ckeditor5) are general ones and only provides `ckeditor.js` with bundled CSS.

**Comment by ostrolucky** [comment]
So are you saying that webpack plugin is responsible for that global state? All I ask is, if possible, extract those sources each time styles are missing when calling `.create()`

**Comment by Reinmar** [comment]
Webpack's `style-loader` works like this – it adds to the bundle a code which, upon its execution, adds  a `<style>` element to the `<head>`. It happens only once. And it's controlled by this loader.

You can check its options (https://github.com/webpack-contrib/style-loader), but I don't think that any of them will resolve your problem automatically.

What you need is to take control over this stylesheet and you can do that by extracting it to a separate file and loading it manually whenever you want.

It's not CKEditor 5's role to handle situations like this where you purge all `<head>`'s content. You need to do it yourself.

**Comment by ostrolucky** [comment]
Fair enough. Thanks for giving me pointers. I just wanted to make sure this is outside of ckeditor control.
